azurerm/resource_arm_virtual_network_gateway.go

```diff
@@ -660,3 +728,35 @@ func validateArmVirtualNetworkGatewayExpressRouteSku() schema.SchemaValidateFunc
 		string(network.VirtualNetworkGatewaySkuTierUltraPerformance),
 	}, true)
 }
+
+func resourceArmVirtualNetworkGatewayCustomizeDiff(diff *schema.ResourceDiff, v interface{}) error {
+
+	if vpnClient, ok := diff.GetOk("vpn_client_configuration"); ok {
+		if vpnClientConfig, ok := vpnClient.([]interface{})[0].(map[string]interface{}); ok {
+			if vpnClientConfig["radius_server_address"] != "" {
+				if !validIP4(vpnClientConfig["radius_server_address"].(string)) {
+					return fmt.Errorf("radius_server_address is not an IPv4 address")
```

Comment:
This validation can be done in the schema, for example:

```go
				ValidateFunc: validation.StringMatch(
					regexp.MustCompile("^[-a-z0-9]{3,50}$"),
					"Cosmos DB Account name must be 3 - 50 characters long, contain only letters, numbers and hyphens.",
				),
```

Reply:
The schema is indeed a better place to do this check.